x-pack/plugins/security_solution/server/endpoint/services/actions/clients/crowdstrike/crowdstrike_actions_client.ts

```diff
@@ -330,6 +306,41 @@ export class CrowdstrikeActionsClient extends ResponseActionsClientImpl {
     return this.fetchActionDetails(actionRequestDoc.EndpointActions.action_id);
   }
 
+  private async completeCrowdstrikeAction(
+    actionResponse: ActionTypeExecutorResult<CrowdstrikeBaseApiResponse> | undefined,
+    doc: LogsEndpointAction
+  ): Promise<void> {
+    const createBaseDocument = (): LogsEndpointActionResponse => ({
+      '@timestamp': moment().toISOString(),
+      agent: doc.agent,
+      EndpointActions: {
+        action_id: doc.EndpointActions.action_id,
+        completed_at: moment().toISOString(),
+        started_at: moment().toISOString(),
+        data: doc.EndpointActions.data,
+        input_type: 'crowdstrike',
+      },
+    });
+
+    const document = createBaseDocument();
+
+    if (actionResponse?.data?.errors?.length) {
+      document.error = {
+        code: '500',
+        message: 'Crowdstrike action failed',
+      };
+    }
+
+    try {
```

Comment:
Please do not do this here. You should have no need to access the underlying index directly from a response actions sub-class.

Please use `this.writeActionResponseToEndpointIndex()` instead

Reply:
Awesome, thanks for reminding me about it :)